### PR TITLE
Add datadog-operator chart

### PR DESCRIPTION
**Pull request by vboulineau**
#### What this PR does / why we need it:

Add chart for `datadog-operator` in our main repository

#### Which issue this PR fixes

Allows to use operator easily

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: charts/datadog-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        app.kubernetes.io/instance: {{ .Release.Name }}
+      annotations:
+        ad.datadoghq.com/{{ .Chart.Name }}.check_names: '["openmetrics"]'
+        ad.datadoghq.com/{{ .Chart.Name }}.init_configs: '[{}]'
+        ad.datadoghq.com/{{ .Chart.Name }}.instances: |
+          [{
+            "prometheus_url": "http://%%host%%:8383/metrics",
+            "namespace": "datadog.operator",
+            "metrics": ["*"]
+          }]
```
Comment (clamoriniere) [inline charts/datadog-operator/templates/deployment.yaml:26]: we should have this as optional until we have a proper "datadog-operator" integration.
Else this metrics will be seen as custom metrics.

File: charts/datadog-operator/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: OPERATOR_NAME
+              value: {{ .Chart.Name }}
+          args:
+            - --zap-level={{ .Values.logLevel }}
+            - --zap-encoder=console
+            # - --zap-stacktrace-level=panic // TODO: uncomment after releasing v0.3.0
```
Comment (clamoriniere) [inline charts/datadog-operator/templates/deployment.yaml:51]: I think we can use this option now.

File: crds/datadoghq.com_datadogmetrics_crd.yaml [hunk 3]
```diff
@@ -0,0 +1,112 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (clamoriniere) [inline crds/datadoghq.com_datadogmetrics_crd.yaml:1]: maybe the installation of this CRD should be optional.

File: charts/datadog-operator/templates/deployment.yaml (hunk 2 above)
Comment (vboulineau) [reply charts/datadog-operator/templates/deployment.yaml:51]: We need to do an operator (docker) release first

File: charts/datadog-operator/templates/deployment.yaml (hunk 1 above)
Comment (vboulineau) [reply charts/datadog-operator/templates/deployment.yaml:26]: Indeed, we should not trigger custom metrics OOB

File: charts/datadog-operator/templates/deployment.yaml (hunk 2 above)
Comment (clamoriniere) [reply charts/datadog-operator/templates/deployment.yaml:51]: I think now it is ok

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**